Assets/Plugins/EuclideonUdSDK/udSDK/csAPI/udSDKAPI.cs

```diff
@@ -197,7 +197,15 @@ unsafe public struct udRenderPicking
         public UInt32 isHighestLOD;//true if hit was as accurate as possible
         public UInt32 modelIndex; //index of the model in the array hit by this pick
         public fixed double pointCenter[3]; //location of the point hit by the pick
-        public IntPtr voxelID; //ID of the hit voxel
+        public udVoxelID voxelID; //ID of the hit voxel
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    public struct udVoxelID
```

Comment:
I am unsure about the correct use of the `unsafe` context, but shouldn't the `udVoxelID` be declared `unsafe` as it has pointer  members? And also does this mean we don't need the `unsafe` keyword for the `udRenderPicking` struct?

Reply:
Unsafe is only necessary when using direct pointer dereferencing- the wrapping types will catch illegal memory dereferencing and throw them as exceptions

Reply:
Yep, and the udRenderPicking class needs to be unsafe not because of the IntPtr but because of the fixed double.